lib/awful/titlebar.lua

```diff
@@ -34,7 +34,8 @@ local all_titlebars = setmetatable({}, { __mode = 'k' })
 -- Get a color for a titlebar, this tests many values from the array and the theme
 local function get_color(name, c, args)
     local suffix = "_normal"
-    if capi.client.focus == c then
+    args.context.has_focus = capi.client.focus == c
```

Comment:
I think the idea here (and your example seems to imply so, too) is that widgets get called with a context that has a `has_focus` entry. However, this does not work.

The function `get_widget_context()` in `wibox.drawable` makes a copy of the `_widget_context_skeleton` that you gave to the `drawable` constructor. This copy will not be updated just because you change the variable here. And even if it were updated, you'd still have to cause a redraw for the whole wibox so that widgets are, well, redrawn with the new context.

Since changing the context implicitly clears the various caches, I don't think that this is a viable approach. Individual widgets already have `context.client`. Adding a `context.has_focus` doesn't *really* add any new information, but causes many issues.